### PR TITLE
Add "ifnewer" as option in help message for pull command

### PR DESCRIPTION
**Pull request by mathiasgredal**
Document the ifnewer pull policy in the help message for the pull command.

Signed-off-by: Mathias Gredal <mathiasgredal@gmail.com>

#### What type of PR is this?

> /kind documentation

#### What this PR does / why we need it:
The if newer pull policy was not documented in the help message for the pull command

#### Which issue(s) this PR fixes:

None
#### Does this PR introduce a user-facing change?
No, the help message the pull command just shows the extra option.

### CODE REVIEW

File: cmd/buildah/pull.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func init() {
 	flags.StringVar(&opts.blobCache, "blob-cache", "", "store copies of pulled image blobs in the specified directory")
 	flags.StringVar(&opts.certDir, "cert-dir", "", "use certificates at the specified path to access the registry")
 	flags.StringVar(&opts.creds, "creds", "", "use `[username[:password]]` for accessing the registry")
-	flags.StringVar(&opts.pullPolicy, "policy", "missing", "missing, always, or never.")
+	flags.StringVar(&opts.pullPolicy, "policy", "missing", "missing, always, ifnewer or never.")
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/pull.go:65]:
```suggestion
	flags.StringVar(&opts.pullPolicy, "policy", "missing", "missing, always, ifnewer, or never.")
```

**Comment by TomSweeneyRedHat** [comment]
@mathiasgredal thanks for the PR.  You'll need to add a `[NO NEW TESTS NEEDED]` tag to your description to make the tests happy.

LGTM, @vrothberg WDYT?

**Review by vrothberg — COMMENTED** [review]
Changes LGTM
@cevich can you check the cross-compile error?

**Comment by TomSweeneyRedHat** [comment]
@mathiasgredal can I ask you to rebase again please (git fetch, rebase, push) ?  We're hoping that will clear up the cross compile test, obviously not an issue that you started with this pr.

**Comment by rhatdan** [comment]
/hold 
/lgtm
Thanks @mathiasgredal

**Review by vrothberg — COMMENTED** [review]
restarted the flaked test

**Comment by rhatdan** [comment]
/hold cancel
Thanks @mathiasgredal

**Pull request merged**